ordersettler/ordersettler.go

```diff
@@ -326,7 +328,18 @@ func (r *OrderSettler) SettleBatch(ctx context.Context, batch types.SettlementBa
 			}
 
 			if rawTx == "" {
-				continue
+				return nil
+			}
+
+			submittedTxs, err := r.db.GetSubmittedTxsByOrderSettlementId(ctx, sql.NullInt64{Int64: settlement.ID, Valid: true})
+			if err != nil {
+				return fmt.Errorf("checking for existing submitted tx for settlement %d: %w", settlement.ID, err)
+			}
+			if len(submittedTxs) > 0 {
```

Comment:
dont submit a tx if one already exists for this settlement

Reply:
hmm this happens after the settlement gets submitted at line 307. this line here should already prevent repeat settlement initiation https://github.com/skip-mev/skip-go-fast-solver/blob/7d45cc6aa45682230e82edd8432be7999a27314b/ordersettler/ordersettler.go#L243-L248

Reply:
i hit a state where the solver was constantly trying to re-submit the tx yesterday, but I can't reproduce right now. do you see any harm in keeping this second check?